spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java

```diff
@@ -55,23 +58,18 @@
 public class KubernetesDiscoveryClientAutoConfiguration {
 
 	@Bean
-	public KubernetesClientServicesFunction servicesFunction(KubernetesDiscoveryProperties properties) {
-		if (properties.serviceLabels().isEmpty()) {
-			if (properties.allNamespaces()) {
-				return (client) -> client.services().inAnyNamespace();
-			}
-			else {
-				return KubernetesClient::services;
-			}
-		}
-		else {
-			if (properties.allNamespaces()) {
-				return (client) -> client.services().inAnyNamespace().withLabels(properties.serviceLabels());
-			}
-			else {
-				return (client) -> client.services().withLabels(properties.serviceLabels());
-			}
+	public KubernetesClientServicesFunction servicesFunction(KubernetesDiscoveryProperties properties,
```

Comment:
this simplifies the code a bit with less branches. You can pass `withLabels` all the time, without the need to check if they are present or not, simply because the default value for `properties.serviceLabels()` is an empty list